### DISCUSSION TITLE
Modbus cannot connect HA 2023.9.2 Integration 2.4.5

### DISCUSSION THREAD

**Discussion by VIMVa** [WillCodeForCats/solaredge-modbus-multi#437] [General]
### Describe the bug

No connection.
Error:
Deze fout is ontstaan door een aangepaste integratie.

Logger: pymodbus.logging
Source: custom_components/solaredge_modbus_multi/hub.py:454
Integration: SolarEdge Modbus Multi (documentation, issues)
First occurred: 16:32:05 (52 occurrences)
Last logged: 17:26:39

Failed to connect [Errno 111] Connect call failed ('192.168.1.36', 1502)
Failed to connect

### Expected behavior

Connection

### Screenshots

![Screenshot_2023-09-13-17-32-05-48_c3a231c25ed346e59462e84656a70e50](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/59361266/17fde8b6-3d69-4250-92b1-6aeaa283277f)

### Diagnostic File

NA

### Debug logs

```shell
Deze fout is ontstaan door een aangepaste integratie.

Logger: pymodbus.logging
Source: custom_components/solaredge_modbus_multi/hub.py:454
Integration: SolarEdge Modbus Multi (documentation, issues)
First occurred: 16:32:05 (52 occurrences)
Last logged: 17:26:39

Failed to connect [Errno 111] Connect call failed ('192.168.1.36', 1502)
Failed to connect
```

### Home Assistant Version

2023.9.2

### solaredge-modbus-multi Version

2.4.5

### Installation Type

HAOS

### Read the Instructions

- [X] I have read https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki
- [X] My configuration follows https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration
- [X] This issue is not about a template error (template errors should not be reported as an issue, read https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Template-Design-Notes)

**Comment by schlurchus** [comment]
Happend to me too. Didn´t get it to work since then. After a restart of the inverter it worked for a few minutes but now again no data.

Thank you

[home-assistant_solaredge_modbus_multi_2023-09-13T20-39-20.911Z.log](https://github.com/WillCodeForCats/solaredge-modbus-multi/files/12601752/home-assistant_solaredge_modbus_multi_2023-09-13T20-39-20.911Z.log)

**Reply by Oht8wooWi8yait9n** [reply]
I'm seeing this as well.  My data stopped on Saturday, September 9th.  I wonder if this is related to a SolarEdge firmware update.

https://www.solaredge.com/en/support/setapp-inverters-firmware#Released32

**Comment by VIMVa** [comment]
That may very well be the case: my inverter did not send it's data to the 'mySolarEdge' platform either between sunrise and this noon 14h30, when I rebooted it to try to reconnect the integration which failed round 12h40.

**Reply by VIMVa** [reply]
I see in mySolarEdge that 'Last update' was yesterday 23h26.
Will check actual version number after a good night's sleep. 😴

**Comment by WillCodeForCats** [comment]
It's working fine for me, so I can't duplicate the issue.

**Reply by Oht8wooWi8yait9n** [reply]
What firmware version is your inverter currently on?

**Reply by WillCodeForCats** [reply]
0004.0014.0229

**Reply by Oht8wooWi8yait9n** [reply]
It looks like I'm actually still on version 4.17.221 (April 2023).  So I guess it's not a firmware update issue.

**Reply by Oht8wooWi8yait9n** [reply]
I just realized I was still on an older version of solaredge-modbus-multi.  After updating to version 2.4.5, I was able to reestablish my Modbus/TCP connectivity.

**Comment by VIMVa** [comment]
I'm on 4.18.32, the version just released.
And, although I'm 100% sure that the Modbus was ON before I reset the converter yesterday, it now was off.
Communication with HA resumed immediately after setting it on.
So no doubt that the loss of connection was caused during and after the firmware update.
Verry sorry to have this launched as an integration issue! But I had checked everything else I was able to. Lesson learned.

**Comment by kino13** [comment]
I also have this issue. It happened the moment I tried to install the two last HA updates. 
What information can I provide for troubleshooting? 
At the moment I had to revert back to HA 9.1 to keep it working. 
edit: 
Device SE5K-RWBTEBEN4
Firmware: 0004.0016.0023 
I am currently on HA 2023.9.0, every time I tried to update it broke the integration and it stopped updating. 
[config_entry-solaredge_modbus_multi-f8946515d3149912683f62768b3645be.json.txt](https://github.com/WillCodeForCats/solaredge-modbus-multi/files/12725504/config_entry-solaredge_modbus_multi-f8946515d3149912683f62768b3645be.json.txt)

**Reply by WillCodeForCats** [reply]
Your version is too old. 2.4.5 is a required update for Home Assistant 2023.9.1 due to changes in pymodbus.

You must update to HA 2023.9.1 or higher before this update will appear in HACS

**Comment by kino13** [comment]
You are... totally right. I had not seen the pending update of
HACS/Modbus-multi after trying the updates.

I have updated all the system components, everything is working as
expected.

Thanks!
